src/hooks/useForm.ts

```diff
@@ -250,11 +255,21 @@ export const useForm = () => {
         fieldsCopy[fieldName].state = 'default';
       }
 
-      onSubmit(transformFormToValues(fields));
-
       setFields(fieldsCopy);
       setHasFormChanged(false);
       setIsSubmitDisabled(true);
+
+      onSubmit(transformFormToValues(fields)).catch((error: AxiosError | PncError) => {
+        if (isAxiosError(error)) {
+          if (error.code === AxiosError.ERR_NETWORK) {
+            setIsSubmitDisabled(false);
+          }
+        } else if (isPncError(error)) {
+          if (error.code === 'NEW_ENTITY_ID_ERROR') {
+            uiLogger.error(error.message);
+          }
+        }
```

Comment:
Missing final `else`, are we sure no other `Error` can happen?

Reply:
Quite much so, but I put there uiLogger logging just in case.